File: ruby/easy/remove_duplicates_from_sorted_array.rb
# @param {Integer[]} nums
# @return {Integer}
def remove_duplicates(nums)
    return 0 if nums.empty?
    i = j = 0
    for j in 0..nums.length-1
      if(nums[i] != nums[j])
        i += 1
        nums[i] = nums[j]
      end
    end
    return i+1
end

File: javascript/easy/removeDuplicatesFromSortedArray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    let i = 0, j = 0;
    for(j = 0; j < nums.length; j++) {
        if(nums[i] != nums[j]) {
            nums[++i] = nums[j];
        }
    }
    return (nums.length == 0) ? 0 : (i+1);
};